ray-operator/apis/ray/v1/rayservice_types.go

```diff
@@ -20,6 +20,13 @@ const (
 	FailedToUpdateService            ServiceStatus = "FailedToUpdateService"
 )
 
+type UpgradeStrategy string
+
+const (
+	BlueGreenUpgrade      UpgradeStrategy = "BlueGreenUpgrade"
```

Comment:
Thinking of alternatives out loud:
1) RollingUpdate / Recreate
2) CreateThenDelete / DeleteThenCreate
3) CreateFirst / DeleteFirst

Reply:
Or perhaps we can follow upgrade strategy types similar to StatefulSet: RollingUpdate / OnDelete https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies

Reply:
I can see "RollingUpdate" being misleading because we only ever update 1 new RayCluster at a time, so it's not really a rolling update

Reply:
I think we can use the "OnDelete" as I think it fits the "delete first" and "delete then create". For the zero downtime maybe we can name it something along the lines of "PrepareUpdateThenSwap" to describe that the updated cluster is getting ready?

Reply:
`OnDelete` implies the users has to manually delete the cluster first. This is a new behavior which is maybe okay? If we are still automatically deleting the cluster, then it should be just `Delete`, `DeleteFirst`, `Replace` something like that.

Reply:
Ahh I see, so like upgrade **OnDelete**. If that's the case, my vote would be for "Replace" or "DeleteFirst".

Reply:
Thinking about this more, I think having a separate strategy for OnDelete and DeleteFirst would be valuable. So in total we could support 3 upgrade strategies
1) spec.upgradeStrategy=RollingUpdate  -- same behavior as ENABLE_ZERO_DOWNTIME=true 
2) spec.upgradeStrategy=DeleteFirst -- same behavior as ENABLE_ZERO_DOWNTIME=false 
3) spec.upgradeStrategy=OnDelete -- new behavior, only upgrade if user manually deletes the RayCluster

Reply:
> OnDelete implies the users has to manually delete the cluster first. This is a new behavior which is maybe okay?

I think the behavior of `OnDelete` is fine, but it may require additional code changes, such as updating the CR status. This is beyond the scope of this PR. Maybe we can open an issue to track the progress, and we can revisit whether we need this behavior after the refactoring of RayService.

Reply:
For `ENABLE_ZERO_DOWNTIME=false` case, how about name it to `None` or `Disabled` or `NoUpgrade`?

Reply:
I think the current behavior of `ENABLE_ZERO_DOWNTIME=false` is to ignore all spec changes and do nothing.